lib/vmdb/console_methods.rb

```diff
@@ -44,13 +44,21 @@ def simulate_queue_worker(break_on_complete: false, quiet_polling: true)
       deliver_on = MiqQueue.arel_table[:deliver_on]
       loop do
         q = with_console_sql_logging_level(quiet_polling ? 1 : ActiveRecord::Base.logger.level) do
-          MiqQueue.where(deliver_on.eq(nil).or(deliver_on.lteq(Time.now.utc)))
-                  .where.not(:queue_name => "miq_server").order(:id).first
+          MiqQueue.where.not(:state => MiqQueue::STATE_DEQUEUE)
+                  .where(deliver_on.eq(nil).or(deliver_on.lteq(Time.now.utc)))
+                  .where.not(:queue_name => "miq_server")
+                  .order(:id)
+                  .first
         end
         if q
           puts "\e[33;1m\n** Delivering #{MiqQueue.format_full_log_msg(q)}\n\e[0;m"
-          q.update!(:state => MiqQueue::STATE_DEQUEUE, :handler => MiqServer.my_server)
-          q.deliver_and_process
+          begin
+            q.update!(:state => MiqQueue::STATE_DEQUEUE, :handler => MiqServer.my_server)
+          rescue ActiveRecord::StaleObjectError
+            puts "\e[33;1m\n** Skipping delivery since it is being processed by another simulator\n\e[0;m"
+          else
+            q.deliver_and_process
+          end
```

Comment:
Maybe not for this PR but I wonder if we could re-use `MiqQueue.get` since this is getting closer and closer to that logic.  Having to handle two simulate_queue_workers is very similar to having to handle two MiqGenericWorkers.

Reply:
This tends towards a rewrite of `MiqQueue.get`.
We introduced `deliver_and_process` to try and remove the duplication.
Interesting that we are introducing yet more from `MiqQueue.get`

Reply:
I guess I'm wondering aside from queue_name/role filtering how does `simulate_queue_worker` differ from a `MiqQueueWorkerBase::Runner`?  I'm not arguing to increase duplication but delete a bunch :)

Reply:
simulate_queue_worker probably doesn't differ from a worker aside from having the inline logging of SQL.  We can definitely look at merging those 2, but I think it should be a different PR.